06-reactjs/components/input.js

```diff
@@ -0,0 +1,37 @@
+var React = require('react');
+var _ = require('lodash');
+
+var Input = React.createClass({
+
+    propTypes: {
+        index: React.PropTypes.string,
+        onChange: React.PropTypes.func,
+        value: React.PropTypes.string
+    },
+
+    getInitialState: function () {
+        var initialState = {};
+
+        if (_.isUndefined(this.props.value)) {
+            initialState.value = '';
+        }
+
+        return initialState;
+    },
+
+    render: function () {
+        return (
+            <input type="text" className="input" value={this.getValue()} onChange={this.handleChange}/>
```

Comment:
Maybe this could be refactored to use a function which returns all this properties